algo

```diff
@@ -208,59 +208,70 @@ EXTRA_VARS="do_access_token=$do_access_token do_server_name=$do_server_name do_r
-  read -p "
-Enter your aws_access_key (http://docs.aws.amazon.com/general/latest/gr/managing-aws-access-keys.html)
-Note: Make sure to use either your root key (recommended) or an IAM user with an acceptable policy attached
-[pasted values will not be displayed]
-[AKIA...]: " -rs aws_access_key
+  if [[ -z $AWS_ACCESS_KEY_ID ]]; then
+    read -p "
+  Enter your aws_access_key (http://docs.aws.amazon.com/general/latest/gr/managing-aws-access-keys.html)
+  Note: Make sure to use either your root key (recommended) or an IAM user with an acceptable policy attached
+  [pasted values will not be displayed]
+  [AKIA...]: " -rs aws_access_key
+  else
+    aws_access_key=$AWS_ACCESS_KEY_ID
+  fi
+
+  if [[ -z $AWS_SECRET_ACCESS_KEY ]]; then
+    read -p "
+  Enter your aws_secret_key (http://docs.aws.amazon.com/general/latest/gr/managing-aws-access-keys.html)
+  Note: Make sure to use either your root key (recommended) or an IAM user with an acceptable policy attached
+  [pasted values will not be displayed]
+  [ABCD...]: " -rs aws_secret_key
+  else
+    aws_secret_key=$AWS_SECRET_ACCESS_KEY
+  fi
 
   read -p "
-Enter your aws_secret_key (http://docs.aws.amazon.com/general/latest/gr/managing-aws-access-keys.html)
-Note: Make sure to use either your root key (recommended) or an IAM user with an acceptable policy attached
-[pasted values will not be displayed]
-[ABCD...]: " -rs aws_secret_key
-
-read -p "
-Name the vpn server:
-[algo]: " -r aws_server_name
-  aws_server_name=${aws_server_name:-algo}
-
-  read -p "
-  What region should the server be located in?
-    1.   us-east-1           US East (N. Virginia)
-    2.   us-east-2           US East (Ohio)
-    3.   us-west-1           US West (N. California)
-    4.   us-west-2           US West (Oregon)
-    5.   ap-south-1          Asia Pacific (Mumbai)
-    6.   ap-northeast-2      Asia Pacific (Seoul)
-    7.   ap-southeast-1      Asia Pacific (Singapore)
-    8.   ap-southeast-2      Asia Pacific (Sydney)
-    9.   ap-northeast-1      Asia Pacific (Tokyo)
-    10.  eu-central-1        EU (Frankfurt)
-    11.  eu-west-1           EU (Ireland)
-    12.  eu-west-2           EU (London)
-    13.  sa-east-1           South America (São Paulo)
-    14.  ca-central-1        Canada (Central)
-Enter the number of your desired region:
-[1]: " -r aws_region
-  aws_region=${aws_region:-1}
-
-  case "$aws_region" in
-    1) region="us-east-1" ;;
-    2) region="us-east-2" ;;
-    3) region="us-west-1" ;;
-    4) region="us-west-2" ;;
-    5) region="ap-south-1" ;;
-    6) region="ap-northeast-2" ;;
-    7) region="ap-southeast-1" ;;
-    8) region="ap-southeast-2" ;;
-    9) region="ap-northeast-1" ;;
-    10) region="eu-central-1" ;;
-    11) region="eu-west-1" ;;
-	12) region="eu-west-2";;
-    13) region="sa-east-1" ;;
-    14) region="ca-central-1" ;;
-  esac
+  Name the vpn server:
+  [algo]: " -r aws_server_name
+    aws_server_name=${aws_server_name:-algo}
+
+  if [[ -z $AWS_DEFAULT_REGION ]]; then
```

Comment:
Let's move this into roles/cloud-ec2/tasks/main.yml in the same format as the keys. I would like to keep the env lookups in the roles themselves rather than the bash script so that there is no dependency on the script for the roles to function.